client/QuickStart/SampleApp.cs

```diff
@@ -0,0 +1,303 @@
+        /// <param name="mappingValue">Values of the new mappings to create</param>
+        /// <param name="dataFormat">Given data format</param>
+        /// <returns>True if Ingestion Mappings exists (whether by us, or the already existing one)</returns>
+        private static async Task<bool> CreateIngestionMappingsAsync(bool useExistingMapping, ICslAdminProvider adminClient, string configDatabaseName, string configTableName, string mappingName, string mappingValue, DataSourceFormat dataFormat)
+        {
+            if (useExistingMapping || mappingValue is null)
+                return true;
+
+            var ingestionMappingKind = dataFormat.ToIngestionMappingKind().ToString().ToLower();
+            WaitForUserToProceed($"Create a '{ingestionMappingKind}' mapping reference named '{mappingName}'");
+
+            mappingName = mappingName ?? "DefaultQuickstartMapping" + Guid.NewGuid().ToString().Substring(0, 5);
+            var mappingCommand = $".create-or-alter table {configTableName} ingestion {ingestionMappingKind} mapping '{mappingName}' '{mappingValue}'";
+
+            try
+            {
+                await Util.ExecuteAsync(adminClient, configDatabaseName, mappingCommand);
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+
+        }
+
+        /// <summary>
+        /// Ingest data from given source.
+        /// </summary>
+        /// <param name="dataSource">Given data source</param>
+        /// <param name="dataFormat">Given data format</param>
+        /// <param name="ingestClient">Client to ingest data</param>
+        /// <param name="configDatabaseName">DB name</param>
+        /// <param name="configTableName">Table name</param>
+        /// <param name="mappingName">Desired mapping name</param>
+        private static async Task IngestDataAsync(IReadOnlyDictionary<string, string> dataSource, DataSourceFormat dataFormat, IKustoIngestClient ingestClient, string configDatabaseName, string configTableName, string mappingName)
+        {
+            var sourceType = dataSource["sourceType"].ToLower();
+            var sourceUri = dataSource["dataSourceUri"];
+            WaitForUserToProceed($"Ingest '{sourceUri}' from '{sourceType}'");
+
+            // Tip: When ingesting json files, if each line represents a single-line json, use MULTIJSON format even if the file only contains one line.
+            // If the json contains whitespace formatting, use SINGLEJSON. In this case, only one data row json object is allowed per file.
+            dataFormat = dataFormat == DataSourceFormat.json ? DataSourceFormat.multijson : dataFormat;
+
+            // Tip: Kusto's C# SDK can ingest data from files, blobs and open streams.See the SDK's samples and the E2E tests in azure.kusto.ingest for additional references.
+            switch (sourceType)
+            {
+                case "localfilesource":
```

Comment:
Maybe make this an enum too?

Reply:
Ok. Do we however want to provide more options then? do we want the user to be able to expand upon it?

Reply:
We don't want to provide anything else right now.